actors/sunbird-lms-mw/actors/common/src/main/java/org/sunbird/learner/actors/bulkupload/UserBulkMigrationActor.java

```diff
@@ -73,19 +84,45 @@ private void processCsvBytes(Map<String, Object> data, Request request) throws I
     Map<String, Object> result = getUserById(userId);
     String channel = getChannel(result);
     String rootOrgId = getRootOrgId(result);
-    List<MigrationUser> migrationUserList =
-        getMigrationUsers(channel, processId, (byte[]) data.get(JsonKey.FILE), values);
-    ProjectLogger.log(
-        "UserBulkMigrationActor:processRecord: time taken to validate records of size "
-                .concat(migrationUserList.size() + "")
-            + "is(ms): ".concat((System.currentTimeMillis() - validationStartTime) + ""),
-        LoggerEnum.INFO.name());
+    Map<String, Object> values = mapper.readValue(systemSetting.getValue(), Map.class);
+
     request.getRequest().put(JsonKey.ROOT_ORG_ID, rootOrgId);
-    BulkMigrationUser migrationUser = prepareRecord(request, processId, migrationUserList);
-    ProjectLogger.log(
-        "UserBulkMigrationActor:processRecord:processing record for number of users "
-            .concat(migrationUserList.size() + ""),
-        LoggerEnum.INFO.name());
+    BulkMigrationUser migrationUser = null;
+    if (request
+        .getOperation()
+        .equals(BulkUploadActorOperation.USER_BULK_SELF_DECLARED.getValue())) {
+      PropertiesCache propertiesCache = PropertiesCache.getInstance();
+      String mandatoryFields = propertiesCache.getProperty(JsonKey.SELF_DECLARED_MANDATORY_FIELDS);
```

Comment:
The mandatory fields are always fixed - status, error-type, orgId, user id.
The optional fields need not be assumed and is irrelevant to be checked. If these are not provided, then processing wouldn't be complete. Of the optional values, presently only 
1. declared-external-id (forget the key, but hope you get the point) is useful
2. declared-school-udise-code 
are used to process the record. If these are not found, record can be marked failed.